pool/allocator.go

```diff
@@ -38,6 +40,9 @@ func NewPoolAllocator(ctx context.Context, timeout time.Duration, factory Factor
 			return nil, err
 		}
 
+		// should be in factory.SpawnWorkerWithTimeout, but this method is missing cfg parameter
+		w.SetMaxExecs(calculateMaxExecs(cfg.MaxJobs, cfg.MaxJobsDispersion))
```

Comment:
No need to use an additional method. You can modify initial `MaxJobs` (if > 0). Use `rand` package to set a some value from range `max_job:max_jobs+10%`. Some constant jitter.

Reply:
cfg.MaxJobs parameter is the same for all workers in the pool, but I want to make it different for each worker

Reply:
Yeah, forgot about that, you're right.